StripePaymentSheet/StripePaymentSheet/PaymentSheet/PaymentSheetFormFactory/FormSpec/FormSpecPaymentHandler.swift

```diff
@@ -120,6 +129,34 @@ extension STPPaymentHandler {
     func _isPIStatusSpecFinished(paymentIntentStatusSpec: FormSpec.NextActionSpec.PostConfirmHandlingPiStatusSpecs) -> Bool {
         return paymentIntentStatusSpec.type == .finished
     }
+
+    func followRedirects(to url: URL, urlSession: URLSession) -> URL {
+        let urlRequest = URLRequest(url: url)
+        let blockingDataTaskSemaphore = DispatchSemaphore(value: 0)
+
+        var resultingUrl = url
+        let task = urlSession.dataTask(with: urlRequest) { data, response, error in
+            guard error == nil,
+                  let httpResponse = response as? HTTPURLResponse,
+                  (200...299).contains(httpResponse.statusCode),
+                    let responseURL = response?.url else {
+                blockingDataTaskSemaphore.signal()
```

Comment:
I like to put the signal calls in a `defer` block at the start of the closure so you don't have to call it on each branch

Reply:
thank you, i need to get in the habit of doing this.